flink/v1.12/flink/src/main/java/com/netease/arctic/flink/read/source/log/kafka/LogKafkaSourceBuilder.java

```diff
@@ -371,6 +387,50 @@ public LogKafkaSource build() {
         tableProperties);
   }
 
+  private void convertArcticProperties() {
+    if (tableProperties.containsKey(LOG_STORE_ADDRESS)) {
+      props.put(BOOTSTRAP_SERVERS_CONFIG, tableProperties.get(LOG_STORE_ADDRESS));
+    }
+    if (tableProperties.containsKey(LOG_STORE_MESSAGE_TOPIC)) {
+      setTopics(getLogTopic(tableProperties));
+    }
+
+    props.putIfAbsent("properties.key.serializer", "org.apache.kafka.common.serialization.ByteArraySerializer");
+    props.putIfAbsent("properties.value.serializer", "org.apache.kafka.common.serialization.ByteArraySerializer");
+    props.putIfAbsent("properties.key.deserializer", "org.apache.kafka.common.serialization.ByteArrayDeserializer");
+    props.putIfAbsent("properties.value.deserializer", "org.apache.kafka.common.serialization.ByteArrayDeserializer");
+
+    setStartupMode();
+  }
+
+  private void setStartupMode() {
```

Comment:
What's the priority of these two methods? setStartupMode and setStartingOffsets. 
I think `setStartingOffsets` has higher priority than this method `setStartupMode`.

besides, the method name setupStartupMode is better.

Reply:
The priority is the same as you think. setStartupMode is only called in the init of Class, and it can be overlaid by setStartingOffsets.
I will rename it.